### PR TITLE
(#5) Add option to specify the user that should run the NRPE checks

### PR DESCRIPTION
**Pull request by jeroen92**
It's pretty self-explanatory I guess :)

We've got this patch enabled a year and a half now, as we don't want the checks to run as the root user. With this change it by default still does, but it allows you to override the `runas_user` in the plugin config.

### CODE REVIEW

File: agent/nrpe.rb [hunk 1]
```diff
@@ -46,17 +46,25 @@ class Nrpe<RPC::Agent
...
       #          plugin.nrpe.conf_file = checks.nrpe
+      #          plugin.nrpe.runas_user = nrpe
       def self.run(command, args=[])
         nrpe_command = Nrpe.plugin_for_command(command, args)
 
         return 3, "No such command: #{command}" unless nrpe_command
 
         output = ""
+        config = Config.instance
+        runas_user = config.pluginconf["nrpe.runas_user"] || nil
```
Comment (ripienaar) [inline agent/nrpe.rb]: `runas_user = @config.pluginconf["nrpe.runas_user"]`, you dont need to set config and dont need the `|| nil` :)

File: agent/nrpe.rb [hunk 2]
```diff
@@ -46,17 +46,25 @@ class Nrpe<RPC::Agent
...
+        unless runas_user.nil?
+          nrpe_command = "sudo -u #{runas_user} #{nrpe_command}"
```
Comment (ripienaar) [inline agent/nrpe.rb]: `nrpe_command = "sudo -u '%s' %s" % [runas_user, nrpe_command] if runas_user`

**Comment by jeroen92** [comment]
Yes, I'm working on the tests, but documentation on how to setup a test environment seems to be lacking. With ruby 2.1.10 running rake throws all sorts of deprecation exceptions.

I'm a bit familiar writing spectests, so I'll probably be able to figure that part out. Do you have any tips for getting a proper development environment up and running for such Choria plugins?

Ty for the improved bits of code, I'll incorporate them into the MR.

**Comment by ripienaar** [comment]
It's via bundler so:

```
% bundle install
...
Bundle complete! 5 Gemfile dependencies, 14 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
% bundle exec rake test
........................

Finished in 0.06416 seconds
24 examples, 0 failures
```

File: agent/nrpe.rb (hunk 1 above)
Comment (jeroen92) [reply agent/nrpe.rb]: @config is not available

Comment (ripienaar) [reply agent/nrpe.rb]: ah doh, its because this is a class method not an instance one - you're right, nevermind me :)

**Comment by jeroen92** [comment]
Wrote some tests but they're not working for some reason.
MR is updated.

**Comment by ripienaar** [comment]
OK, I'll see if i can fix your tests for you, no biggie

**Comment by jeroen92** [comment]
@ripienaar Thanks a lot! Spectests sometimes still feel like a big incantation to me :)
I've squashed in your changes, all seems to be working fine now :)

**Pull request merged**